UI/ViewModel/Base/BaseItemViewModel.cs

```diff
@@ -178,6 +193,28 @@ public void Apply()
                 {
                     Update();
                 }
+                catch (FaultException<TodoClient.ForeignKeyConstraintFault> ex)
```

Comment:
FaultException has meaningless standard message. WCF-client rethrows a new FaultException exception with the received data.  It is a oppotunity to include message property into the recieved *Fault object but to retrieve this message we need use a reflection or adapter or visitor.
I see two ways to decide this problem: 
1) make prism-like error aggregator 
2) make model layer thicker and move part of VM logic into M to avoid the anemic client domain logic.